cmake/gen_version_h.cmake

```diff
@@ -16,6 +16,9 @@ if(NOT DEFINED ${BUILD_VERSION_NAME})
   git_describe(${work_dir} ${BUILD_VERSION_NAME})
 endif()
 
+git_commit_hash(${work_dir} GIT_COMMIT_HASH)
+git_commit_hash_short(${GIT_COMMIT_HASH} GIT_COMMIT_HASH_SHORT GIT_COMMIT_HASH_SHORT_UINT32)
```

Comment:
this code assumes that previous call (`git_commit_hash()`) correctly populated the `GIT_COMMIT_HASH` with a single value. But it is exactly in cases where this unexpectedly doesn't happen that can make CMake debugging hard, cause the error you get is:
```
CMake Error at gen_version_h.cmake.20 (git_commit_hash_short):                                                                                                                                                                              
  git_commit_hash_short Function invoked with incorrect arguments for function named: git_commit_hash_short 
```
so you never get to your function validation, and a user seeing such error may not realize what is wrong.

Quoting `"${GIT_COMMIT_HASH}"` can help a little as that at least will ensure that the argument expands to an empty string (and later function validation can work), instead of expanding to nothing.

Guess it's debugging surprises like this @marc-hb refers to :slightly_smiling_face: